Tribler/Category/Category.py

```diff
@@ -89,14 +90,8 @@ def calculateCategory(self, torrent_dict, display_name):
         return self.calculateCategoryNonDict(files_list, display_name, tracker, comment)
 
     def calculateCategoryNonDict(self, files_list, display_name, tracker, comment):
-        # Check xxx
-        try:
-
-            if self.xxx_filter.isXXXTorrent(files_list, display_name, tracker, comment):
-                return 'xxx'
-        except:
-            self._logger.critical(
-                'Category: Exception in explicit terms filter in torrent: %s', display_name, exc_info=True)
```

Comment:
Here it would return None if an exception was raised. Have you checked for that on all the callers of this fn?